src/resolvers/projectResolver.ts

```diff
@@ -235,6 +238,24 @@ export class ProjectResolver {
       { category },
     );
   }
+
+  static addMainCategoryQuery(
+    query: SelectQueryBuilder<Project>,
+    mainCategory: string,
+  ) {
+    if (!mainCategory) return query;
+
+    query = query
+      .leftJoinAndSelect('project.categories', 'categoryForMainCategorySearch')
+      .innerJoinAndSelect(
```

Comment:
TypeOrm only serializes columns or virtual columns that exist in the entity, so this attributes categoryForMainCategorySearch  and filteredMainCategory, won't be selectable. However the ```JoinAndSelect``` is attempting to load the data. 
We should use ```leftJoin```and ```ìnnerJoin```to avoid extra work.

Reply:
Test Pass with those too!

Reply:
Thanks @CarlosQ96 , I didn't know that, thanks for finding and fixing this